### PR TITLE
Added cleanup step after deep_integration_tests and custom_domain_integration_tests

### PR DESCRIPTION
**Pull request by alpeb**
Also temporarily enabled cloud integration tests for this PR to see if does fix those tests.

### CODE REVIEW

File: bin/_test-run.sh [hunk 1]
```diff
@@ -62,11 +62,13 @@ function deep_integration_tests() {
 
     run_test "$(go list $test_directory/.../...)" --linkerd-namespace=$linkerd_namespace
     exit_on_err "error during deep tests"
+    cleanup
 }
 
 function custom_domain_integration_tests() {
```
[the remark this answers is not in the document]
Comment (alpeb) [reply bin/_test-run.sh]: Agreed, that was missing. I'll call that test from `/bin/test-run`.

[the remark this answers is not in the document]
Comment (alpeb) [reply bin/_test-run.sh]: I've just realized why custom domain tests were not included; they require the cluster to be setup from the get go with that custom domain, and for the cloud_integration test we just use an existing cluster.

**Comment by siggy** [comment]
A note on the `docker_deploy` step failing on `bin/docker-push-deps`. For some reason the `~/.ssh/known_hosts` was either not being generated or getting blown away at some point. A couple of debug commands confirm the file is empty, and subsequent ssh commands fail because of it:
```bash
$ cat ~/.ssh/known_hosts
$ ssh -v github@$DOCKER_ADDRESS ls -al
...
debug1: read_passphrase: can't open /dev/tty: No such device or address
```

Full log: https://github.com/linkerd/linkerd2/pull/3730/checks?check_run_id=304493671#step:6:47

**Comment by siggy** [comment]
Merged #3730.

**Pull request closed without merging**